ArduPlane/system.cpp

```diff
@@ -230,6 +230,28 @@ bool Plane::set_mode(Mode &new_mode, const ModeReason reason)
     Mode &old_mode = *control_mode;
     const ModeReason previous_mode_reason_backup = previous_mode_reason;
 
+    // cancel inverted flight
+    auto_state.inverted_flight = false;
+
+    // don't cross-track when starting a mission
+    auto_state.next_wp_crosstrack = false;
+
+    // reset landing check
+    auto_state.checked_for_autoland = false;
```

Comment:
please go over all these changes to system.cpp and either justify why the changes are needed for this PR or remove them

Reply:
willdo.